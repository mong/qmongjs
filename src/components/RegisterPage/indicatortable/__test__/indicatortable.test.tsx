import { BrowserRouter } from "react-router-dom";
import { QueryClient, QueryClientProvider } from "react-query";
import { render, screen, waitFor } from "@testing-library/react";

import opts from "../../../../dev-tools/data/unitnames.json";
import { IndicatorTable } from "../";
import { RegisterNames } from "../../.";

const registerInfo: RegisterNames[] = [
  {
    id: 1,
    rname: "hoftebrudd",
    full_name: "Nasjonalt hoftebruddregister",
    caregiver_data: 1,
    resident_data: 0,
    dg_data: 1,
  },
];
it("registry table renders correctly for a single registry", async () => {
  const props = {
    colspan: 2,
    selection_bar_height: 0,
    legend_height: 0,
    optstu: opts.opts_tu,
    unitNames: ["Nasjonalt"],
    registerNames: registerInfo,
    treatmentYear: 2019,
    medicalFieldFilter: ["hoftebrudd"],
  };
  const queryClient = new QueryClient();

  const { container } = render(
    <QueryClientProvider client={queryClient}>
      <BrowserRouter>
        <IndicatorTable
          context={"sykehus"}
          tableType="singleRegister"
          {...props}
        />
      </BrowserRouter>
    </QueryClientProvider>
  );

  await waitFor(() => screen.findAllByRole("heading"));
  expect(container).toMatchSnapshot();
});
