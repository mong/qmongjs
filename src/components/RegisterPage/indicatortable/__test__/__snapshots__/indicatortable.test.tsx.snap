// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`registry table renders correctly for a single registry 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th
          class="quality_indicator"
          style="width: 60%; top: 0px;"
        >
          Kvalitetsindikator
        </th>
        <th
          class="selected_unit"
          style="width: 40%; top: 0px;"
        >
          Nasjonalt
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
